database/migrations/2022_11_17_153635_add_missing_to_mismatches_review_status.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\App;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Schema;
+
+class AddMissingToMismatchesReviewStatus extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        if (App::environment() == 'testing') {
+            // SQLite doesn't support renaming or dropping columns,
+            // so for testing we re-do the entire table
+            $this->dropAndRecreateMismatchesTable();
+        } else {
+            // Laravel doesn't support updating enum columns,
+            // so we modify the column with raw SQL
+            $this->modifyColumn();
+        }
+    }
+
+    /**
+     * Reverse the migrations.
+     *
+     * @return void
+     */
+    public function down()
+    {
+        if (App::environment() == 'testing') {
+            $this->dropAndRestoreMismatchesTable();
+        } else {
+            $this->unmodifyColumn();
+        }
+    }
+
+    private function dropAndRecreateMismatchesTable()
+    {
+        Schema::table('mismatches', function (Blueprint $table) {
+            $table->drop();
+        });
+
+        Schema::create('mismatches', function (Blueprint $table) {
+            $table->id();
+            $table->string('statement_guid');
+            $table->string('item_id');
+            $table->string('property_id');
+            $table->string('wikidata_value', 2048); // Arbitrary length above 1500 chars, upper limit for wikidata value
+            $table->string('external_value', 2048); // Arbitrary length above 1500 chars, upper limit for external value
+            $table->string('external_url', 2048)->nullable(); // Arbitrary length above 1500 chars, upper limit for url
+            $table->enum('review_status', [
+                'pending',
+                'wikidata',
+                'external',
+                'both',
+                'none',
+                'missing'
+            ])->default('pending');
```

Comment:
I think that explicit default is new compared to https://github.com/wmde/wikidata-mismatch-finder/blob/main/database/migrations/2021_08_04_102448_recreate_mismatches_table_with_item_id.php ? But makes sense to me, not something to change

Reply:
I mainly copied this from https://github.com/wmde/wikidata-mismatch-finder/blob/main/database/migrations/2021_08_18_121955_rename_mismatch_status.php (looks like I forgot to mention that in the commit message), where the default was already there.

Reply:
d’oh, and “I copied this from $oldMigration” of course means that the `meta_wikidata_value` is missing, which is why so many tests are failing :facepalm:

Reply:
seems to work now :+1: